### ISSUE TITLE
SVG Coming Through Invisible Unless Mangled 

### ISSUE THREAD

**Issue by victorstewart** [SVGKit/SVGKit#523]
I'm not using cocoapods here. I added the SVG 2.x project to my project.

I'm definitely an SVG Noob, so I'm not sure if this is covered by another post, sorry guys. I'll make up for it with info though lol.

Trying to display this SVG:
https://www.kochind.com/assets/images/logo_kii_color.svg

If you see the commented out line, when I don't include that, the image is invisible when the view is displayed. (It's happening for some SVGs, but not for others).

Whether the commented line is active or not, both log lines print:

_ImageCell - prepare() -> (before) svgImage.size = {115, 17.200000762939453}
ImageCell - prepare() -> (after) svgImage.size = {115, 17.200000762939453}_

And when I do add that line (simply setting its size equal to its true size), the image is displayed, but mangled / stretched.. (see image link below).

![img_0220 png](https://cloud.githubusercontent.com/assets/17534897/25254030/88f78400-25f1-11e7-824e-9121d4da3c8e.png)

Is it something I'm doing wrong? Lemme know if you need any more information.

```
#if DEBUG
        LOG(@"ImageCell - prepare() -> (before) svgImage.size = %@", NSStringFromCGSize(self.svgImage.size));
#endif
 
       self.svgView = [[SVGKFastImageView alloc] initWithSVGKImage:self.svgImage];
      // self.svgView.image.size = self.svgImage.size; /// THIS LINE HAS TO BE HERE
       self.svgView.frame = CGRectMake(0, 0, self.contentView.frame.size.width,      self.contentView.frame.size.height);
        [self.contentView addSubview:self.svgView];
        
#if DEBUG
        LOG(@"ImageCell - prepare() -> (after) svgImage.size = %@", NSStringFromCGSize(self.svgView.image.size));
#endif
```

**Comment by adamgit** [comment]
In the xcode debugger, what changes before/after your commented out line
runs?

**Comment by adamgit** [comment]
NB: explicitly setting an SVGKImage's size *overrides* the real size with
your enforced size. If the original had a size (not all SVG's have a valid
one), SVGKit will resize the SVG.

But it will now store your supplied size, instead of using the internally
calculated one.

**Comment by victorstewart** [comment]
you mean the debugging output I'm printing? Nothing changes. It's exactly the same whether the commented line is in or out.

right I assumed it did that. But the enforced size is equal to the real size, as you see in the code. And that size is short and wide (115 wide, 17 tall). But it's clearly being stretched to fill the container.

I also check the size of `self.svgView.image.size` AFTER it's been displayed on the screen, and it still reads (115 wide, 17 tall)... even though it's clearly not that aspect ratio anymore.

**Comment by adamgit** [comment]
No, I mean the debugger. If you're not using the debugger ... I'm not sure
how you can sensibly debug an iOS project? If you use log statements for
everything, it'll take you literally thousands of times longer to do simple
debugging.

Also: SVG's have a built-in concept of aspect-ratio. Unlike bitmaps, the
aspect ratio can be embeded in the image. It might be that something is
throwing-off the aspect-ratio code - but you'd find this easier to view in
the debugger IMHO.

**Comment by victorstewart** [comment]
Oh, well I built my own logging system into the app. I've never really faced any limitations with it? And I've built some decently complex things.

What debugging output with you like? I'll procure it.

Like this area? before and after?
https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/debugging_with_xcode/Art/dwx-dt-varlist-simple_2x.png

**Comment by adamgit** [comment]
If you click those arrows, you can inspect everything in memory.

Read a guide on it - you can set alerts on variables, have it auto-pause
when specific variables change, inspect exactly what's going on in memory,
etc.

For problems like this - where something is changing in a way you don't
expect - it makes the process of figuring out both "how" and "why" many
times faster. Then we can look at the source and use that info to quickly
decide what to change :).

Off the top of my head, the behaviour you expect sounds reasoanble, so I
have no idea why it's not behaving that way - but I might be forgetting
something about how/what is happening, and the debugger is a quick way for
you to look at it and see for yourself what's going on.

TL;DR: debugger is awesome (it was crappy 7 years ago, it was OK 4 years
ago, but it's really really good today)

**Comment by victorstewart** [comment]
Haha, I'm self-taught, started programming a year ago. First grinded my debugging chops with a C++ server, so I learned all my logging habits from there -- where it's the only option.

I've seen that area before and poked around with it. But I zero-ed it out for all-console ages ago lol.

I'll post those outputs once i capture them

**Comment by adamgit** [comment]
I really, really, really strongly recommend using the debugger. It's like
being in the Matrix, and seeing everything, when compared to using plain
log statements.

Using log statements is a valauble skill to have for extreme siutations.
But it was already outdated and a bad use of your time 20 years ago. Today,
it's far more valuable a skill to know the features of the debugger.

(But even without learning any of the advanced stuff, the debugger
immensely speeds-up development)

**Comment by victorstewart** [comment]
my breakpoints aren't triggering, so I can't capture that output. I tried googling the standard issues why but none solved it. I'll get it working, but in the meanwhile... any other information I can provide? 

Even with log statements lol....

**Comment by adamgit** [comment]
We have a standard image that tests how SVG handles scaling - scale-to-fit,
scale-to-aspect-ratio, etc. You could try opening that - if its broken,
that suggests someone has recently committed something to the 2.x branch
that broke all scaling :).

This image (included in the SVG demo app) -
https://www.w3.org/TR/SVG/images/coords/PreserveAspectRatio.svg

Might be worth reading that section of the spec, too - could be as simple
as turnging on the attribute in your SVG? -
https://www.w3.org/TR/SVG/coords.html#PreserveAspectRatioAttribute

**Comment by victorstewart** [comment]
awesome! thank you. I'll try those shortly
